### ISSUE TITLE
usermod: group '1004' does not exist

### ISSUE THREAD

**Issue by romarknmsu** [PacktPublishing/Practical-Jenkins-2.x#4]
From the book (chapter 2) example, I'm running the command to build the image `docker build -t jenkins/jenkins:2.361.4-lts -f Dockerfile .` It has worked just fine for over a year. I recently upgraded docker and now I see this error below. I'm running this on an internal VM which has the uid and gid of 1003 and 1004. Has anyone else reported this issue? I'm not sure what to do to fix this. I'm running ubuntu 20.04 ubuntu on an internal vm with the latest version of Docker version 23.0.3, build 3e7cbfd.
```
[+] Building 1.4s (5/6)                                                                                                                                                           
 => [internal] load build definition from Dockerfile                                                                                                                         0.1s
 => => transferring dockerfile: 179B                                                                                                                                         0.0s
 => [internal] load .dockerignore                                                                                                                                            0.0s
 => => transferring context: 2B                                                                                                                                              0.0s
 => [internal] load metadata for docker.io/jenkins/jenkins:2.361.4-lts                                                                                                       0.2s
 => CACHED [1/3] FROM docker.io/jenkins/jenkins:2.361.4-lts@sha256:4283a1d1ce2f7893d083ec413d1c196edfddc3fe6e0e9ecfefcfbe99bf3bff9d                                          0.0s
 => ERROR [2/3] RUN usermod -u 1003 -g 1004 jenkins                                                                                                                          1.0s
------                                                                                                                                                                            
 > [2/3] RUN usermod -u 1003 -g 1004 jenkins:
#0 0.936 usermod: group '1004' does not exist
------
Dockerfile:3
--------------------
   1 |     FROM jenkins/jenkins:2.361.4-lts
   2 |     USER root   
   3 | >>> RUN usermod -u 1003 -g 1004 jenkins
   4 |     RUN mkdir /backup && chown 1003:1004 /backup
   5 |     USER jenkins
--------------------
ERROR: failed to solve: process "/bin/sh -c usermod -u 1003 -g 1004 jenkins" did not complete successfully: exit code: 6
```
Here is my dockerfile:

```
FROM jenkins/jenkins:2.361.4-lts
ARG j_uid=1003
ARG j_gid=1004
USER root   
RUN usermod -u ${j_uid} -g ${j_gid} jenkins
RUN mkdir /backup && chown ${j_uid}:${j_gid} /backup
USER jenkins
```

**Comment by calvinpark** [comment]
Hi Mark,

The error is this:
```
 > [2/3] RUN usermod -u 1003 -g 1004 jenkins:
#0 0.936 usermod: group '1004' does not exist
```

I'm realizing that the book's example `RUN  usermod -u 123 -g 30 jenkins` works because group 30 already exists in the `jenkins/jenkins` container. In your case, it would work if you create a group with its ID `1004` in the container before calling `usermod`.

You mentioned that this had worked in the past. Do you know which Jenkins container it was working with? Dockerfiles are static for the most part, so it should be easily reproducible. It could be that the past Jenkins containers had a group with that ID and was recently removed.

**Comment by romarknmsu** [comment]
Hi Calvin, 
Thanks for the reply. The short story is I got it working by going back to docker version `5:20.10.18~3-0~ubuntu-focal`. For sanity and to make sure it was not the jenkins images, I killed the container running `jenkins/jenkins:2.361.4-lts` and upgraded to `jenkins/jenkins:2.387.2-lts` I guess this tells me it's the version of docker that's restricting or preventing the docker build command to fully build. I'll have to look at the release notes of docker to see if something is listed there to get to the bottom of this. I'll post here as I find out more. 

Here is my Dockerfile:

```bash
FROM jenkins/jenkins:2.387.2-lts
USER root   
RUN usermod -u 1003 -g 1004 jenkins
USER jenkins
```
I ran this command to build the image:
`docker build -t jenkins/jenkins:2.387.2-lts -f Dockerfile .`

I ran this to run the container:
`docker run --detach --restart on-failure -u $(id -u):$(id -g) -v ~/jenkins_home:/var/jenkins_home -p 8080:8080 -p 50000:50000 --name jenkins_controller jenkins/jenkins:2.387.2-lts`
